src/components/MediaUploadZone.tsx

```diff
@@ -0,0 +1,148 @@
+import { UploadSimple, WarningCircle } from "@phosphor-icons/react";
+import { useState, useEffect, useRef } from "react";
+
+interface MediaUploadZoneProps {
+  onFilesDropped: (files: File[]) => void;
+}
+
+const MediaUploadZone: React.FC<MediaUploadZoneProps> = ({
+  onFilesDropped,
+}) => {
+  // parent ref
+  const parentRef = useRef<HTMLDivElement>(null);
+
+  // handling sizing
+  const [margin, setMargin] = useState("1em");
+  const [fontSize, setFontSize] = useState("1em");
+  // handling drop in file
+  const [isDragOver, setIsDragOver] = useState(false);
+  const [droppedFiles, setDroppedFiles] = useState<File[]>([]);
+  const [isValidFile, setIsValidFile] = useState(true);
+
+  const UploadZoneClassName =
+    "w-full h-full border-2 border-dashed flex flex-col justify-center items-center rounded-md border-primary-main " +
+    (isDragOver
+      ? " transform transition-transform duration-300 border-primary-dark ease-in-out scale-95"
+      : "");
+
+  // adjust font size according to parent ref
+  useEffect(() => {
+    if (parentRef.current) {
+      const { width, height } = parentRef.current.getBoundingClientRect();
+      const newSize = Math.min(width, height) / 15;
+      setFontSize(`${newSize}px`);
+      setMargin(`${newSize / 10}px  `);
+      width > 700
+        ? setMargin(`${newSize / 10}px`)
+        : setMargin(`${-newSize / 2}px`);
+      console.log(width);
+    }
+  }, []);
+
+  const handleDrop = (event: React.DragEvent<HTMLDivElement>) => {
```

Comment:
Good job on implementing the functionality, but this out of the scope for this ticket. Make sure to clarify this for further tickets.